### PR TITLE
SImplify "astro add" by removing confusing multi-select

### PR DESCRIPTION
**Pull request by FredKSchott**
## Changes

- Our "astro add" multi-select had been causing users trouble.
- This change solves the usability issues by just removing the multi-select flow.
- The multi-select was a neat idea, but this new static output contains all the same information without the usability issues. Plus, it teaches you how to use the real CLI command.
- The analogy that I have come to think of this as: "npm doesn't have a multi-select for 'npm install', and they do just fine, so why should we?" 

**Bonus!** `create-astro` also updated with a few style changes and bug fixes.
**Bonus!** Added link to our full integration catalog.  
**Bonus!** Cleaned up the `printHelp()` function to be a bit more pretty across all the commands, and updated its API to be a bit more flexible.

### Screenshot: `astro add` multi-select replaced with this output
Screen Shot 2022-06-25 at 11 33 16 PM

### Recording: The new, updated `create-astro` flow

https://user-images.githubusercontent.com/622227/175868477-ea018d75-3109-4966-ae69-20a2ff5464a6.mov

## Testing

- Tested manually (`astro add` isn't tested yet)
- This mainly removes logic, so bug risk should be lower than normal

## Docs

- Docs PR created: https://github.com/withastro/docs/pull/867

### CODE REVIEW

**Comment by hippotastic** [comment]
Does the command specifying multiple integrations to add in one call? If so, that would be great, and it should be reflected in the help output somehow.

So rather than `astro add [integration]` we could write something like `astro add [integration names]`, `[integrations...]` or anything else that you see fit.

What I also currently find a bit misleading is that you're introducing astro add in the usage line as a command to install `[integrations]`, but then provide two lists: One of `UI Frameworks` (huh?), and one of `Integrations`. New developers could be confused if UI Frameworks are something else than integrations.

I'd personally also be confused why those integrations are all "recommended". Does that mean I should install all of them?

I'd therefore propose to change the list headers to something like `Example: Add UI Framework Support` and `Example: Add Other Popular Integrations`.

**Comment by delucis** [comment]
Hippo's ideas seem smart — plus one on those.

Re: docs — yes we describe the current CLI output in detail in our install guide, which will need updating.

**Comment by FredKSchott** [comment]
Agreed! Updated copy based on @hippotastic's comments.

> Re: docs — yes we describe the current CLI output in detail [in our install guide](https://docs.astro.build/en/install/auto/#install-any-official-astro-integrations-optional), which will need updating.

Great! PR to update created: https://github.com/withastro/docs/pull/867

> In fact, how will this work as part of create-astro? Do we just remove the add step and recommend running it when the setup CLI is done?

Good call out! I updated the PR to update create-astro for this change. Video preview added to OP.

File: packages/astro/src/core/logger/node.ts [hunk 1]
```diff
@@ -126,6 +126,7 @@ export const logger = {
 
 export function enableVerboseLogging() {
 	//debugPackage.enable('*,-babel');
+	console.log('ah!');
```
Comment (delucis) [inline packages/astro/src/core/logger/node.ts:129]: Was this left over by accident?

```suggestion
```

File: packages/create-astro/src/index.ts [hunk 2]
```diff
@@ -233,30 +211,40 @@ export async function main() {
...
-	console.log(`\nStuck? Come join us at ${bold(cyan('https://astro.build/chat'))}`);
+	await logAndWait(
+		`Add frameworks like ${bold(cyan('react'))} and ${bold(
+			cyan('tailwind')
+		)} to your project using ${bold(cyan('astro add'))}`
+	);
+	await logAndWait('');
+	await logAndWait(`Stuck? Come join us at ${bold(cyan('https://astro.build/chat'))}`, 1000);
+	await logAndWait(dim('Good luck out there, astronaut.'));
+	await logAndWait('', 300);
```
Comment (delucis) [inline packages/create-astro/src/index.ts:247]: Nice touch with the staggered output!

File: packages/create-astro/test/install-step.test.js [hunk 3]
```diff
@@ -26,6 +26,7 @@ describe('[create-astro] install', function () {
 			const seen = new Set();
 			const installPrompt = PROMPT_MESSAGES.install(FAKE_PACKAGE_MANAGER);
 			stdout.on('data', (chunk) => {
+				console.log(chunk.toString());
```
Comment (delucis) [inline packages/create-astro/test/install-step.test.js:29]: Looks like we need to disable ESLint for this line:

```suggestion
				// eslint-disable-next-line no-console
				console.log(chunk.toString());
```

Comment (delucis) [reply packages/create-astro/test/install-step.test.js:29]: Or was it just for debugging and should be removed?

```suggestion
```

**Review by delucis — COMMENTED** [review]
I’m a big plus one on these changes. The multi-select was also really hard to describe in docs, so this will make our life easier 😅

Spotted a couple of tiny things, but otherwise LGTM!

File: .changeset/perfect-drinks-fold.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+'astro': patch
+---
+
+Update the help output to improve formatting
```
Comment (natemoo-re) [inline .changeset/perfect-drinks-fold.md:5]: These changelogs can be combined

File: packages/create-astro/src/index.ts [hunk 5]
```diff
@@ -64,7 +72,7 @@ export async function main() {
 		const dirResponse = await prompts({
 			type: 'text',
 			name: 'directory',
-			message: 'Where would you like to create your app?',
+			message: 'Where would you like to create your new project?',
```
Comment (natemoo-re) [inline packages/create-astro/src/index.ts:75]: 👍🏻

File: packages/astro/src/core/logger/node.ts (hunk 1 above)
Comment (FredKSchott) [reply packages/astro/src/core/logger/node.ts:129]: ah!

File: packages/create-astro/src/index.ts (hunk 2 above)
Comment (FredKSchott) [reply packages/create-astro/src/index.ts:247]: Thanks! I was kind of just playing around at first but it feels a lot more human.

The `1000` wait before "good luck out there" is a bit of a cheesy touch, so if anyone has any thoughts I'm okay to scrap it or shrink it down to a smaller wait. You can see it in action in the video above.

File: .changeset/perfect-drinks-fold.md (hunk 4 above)
Comment (FredKSchott) [reply .changeset/perfect-drinks-fold.md:5]: They're both individually different features impacting different packages (for example, `create-astro` didn't receive a change to "update the help output"). 

I don't feel too strongly, but given that this is more correct I'd argue to keep as-is.

File: packages/create-astro/src/index.ts [hunk 6]
```diff
@@ -233,30 +211,40 @@ export async function main() {
...
-	console.log(
+	await logAndWait(
 		`Run ${bold(cyan(devCmd))} to start the Astro dev server. ${bold(cyan('CTRL-C'))} to close.`
 	);
-	if (!installResponse.install) {
-		console.log(yellow(`Remember to install dependencies first!`));
-	}
-	console.log(`\nStuck? Come join us at ${bold(cyan('https://astro.build/chat'))}`);
+	await logAndWait(
+		`Add frameworks like ${bold(cyan('react'))} and ${bold(
```
[the remark this answers is not in the document]
Comment (natemoo-re) [reply packages/create-astro/src/index.ts:240]: I'm fine with that reasoning! Can leave as is.

**Pull request merged**